### DISCUSSION TITLE
Beats cannot connect, nothing listening on port 5044

### DISCUSSION THREAD

**Discussion by lonewaffle** [Security-Onion-Solutions/securityonion#3863] [Unsupported Versions] [answered]
I'm evaluating SO and am having trouble getting WinLogBeats to connect to it. I've run the so-allow to open up communications with my internal network, installed the beat on a Windows server and the log is showing the following message:

ERROR	[publisher_pipeline_output]	pipeline/output.go:154	Failed to connect to backoff(async(tcp://10.0.0.95:5044)): dial tcp 10.0.0.95:5044: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

The Windows server can ping SO. I've run netstat on SO and don't see anything listening on port 5044. I've seen numerous other reports of this that eventually just started working on it's own so I left it overnight but no joy. I've also confirmed with so-status that everything is running.

Any ideas on what else to check?

**Comment by jcastillo725** [comment]
Have you run so-allow for logstash on the IP range of the endpoints?

**Reply by lonewaffle** [reply]
Yeah, I ran the so-allow for the logstash specifically twice. The second time it complained about it already having the relevant entry but I did it anyway just in case it didn't take the first time. It's my understanding that I should still see port 5044 listening in netstat even if the firewall is blocking it but I don't even see that, so I'm not sure what to look at next.

**Comment by defensivedepth** [comment, marked as answer]
Please follow the support guidelines, specifically the `Provide sufficient technical info`: (It will help us help you much quicker!)

https://github.com/Security-Onion-Solutions/securityonion/discussions/1720#discussion-28649

**Reply by lonewaffle** [reply]
Sorry about that. Here's some technical info:

- Version is 2.3.40
- Installed using the ISO
- Install type is EVAL
- so-status shows everything as running
- Running "salt-call state.highstate", I see the following errors:
[ERROR   ] Could not cache minion ID: [Errno 13] Permission denied: '/etc/salt/minion_id'
[WARNING ] Failed to open log file, do you have permission to write to /var/log/salt/minion?
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds
[ERROR   ] DNS lookup or connection check of 'salt' failed.

**Reply by lonewaffle** [reply]
Nevermind on the salt-call errors. I was dumb and didn't run it as sudo. Running it with the appropriate permissions, it completes without any failures.

**Reply by defensivedepth** [reply]
Thanks! I see that your install type is EVAL.

From the docs: https://docs.securityonion.net/en/2.3/beats.html#beats

`In order to receive logs from Beats, Security Onion must be running Logstash. Evaluation Mode and Import Mode do not run Logstash, so you’ll need Standalone or a full Distributed Deployment.`

**Reply by lonewaffle** [reply]
I suspected that EVAL might have something to do with this. Thanks!
